src/main/scala/rise/core/types/Solution.scala

```diff
@@ -141,18 +139,18 @@ case class Solution(ts: Map[Type, Type],
     case FragmentTypeConstraint(a, b) => FragmentTypeConstraint(apply(a), apply(b))
     case NatToDataConstraint(a, b) => NatToDataConstraint(apply(a), apply(b))
     case NatCollectionConstraint(a, b) => NatCollectionConstraint(apply(a), apply(b))
-    case DepConstraint(NatKind, df, arg: Nat, t) => DepConstraint(NatKind, apply(df), apply(arg), apply(t))
-    case DepConstraint(DataKind, df, arg: DataType, t) =>
+    case DepConstraint(_, df, arg: Nat, t) => DepConstraint(NatKind, apply(df), apply(arg), apply(t))
```

Comment:
Is there a reason why you removed this?

Reply:
No, thanks for catching this, must have gone by in the many edits

Reply:
Restored